src/analyzer/analyzergain.cpp

```diff
@@ -25,29 +26,41 @@ bool AnalyzerGain::initialize(TrackPointer tio, int sampleRate, int totalSamples
     if (loadStored(tio) || totalSamples == 0) {
         return false;
     }
-    m_bStepControl = m_pReplayGain->initialise((long)sampleRate, 2);
+
+    m_initalized = m_pReplayGain->initialise((long)sampleRate, 2);
     return true;
 }
 
 bool AnalyzerGain::loadStored(TrackPointer tio) const {
+    int version = m_pConfig->getValueString(
+            ConfigKey("[ReplayGain]", "ReplayGainVersion")).toInt();
     // WARNING: Do not fix the "analyser" spelling here since user config files
     // contain these strings.
-    bool bAnalyzerEnabled = (bool)m_pConfigReplayGain->getValueString(
-        ConfigKey("[ReplayGain]","ReplayGainAnalyserEnabled")).toInt();
-    if (tio->getReplayGain().hasRatio() || !bAnalyzerEnabled) {
-        return true;
+    bool analyzerEnabled = ((bool)m_pConfig->getValueString(
+            ConfigKey("[ReplayGain]", "ReplayGainAnalyserEnabled")).toInt()) &&
+            (version == 1);
+    bool reanalyse = m_pConfig->getValueString(
+            ConfigKey("[ReplayGain]", "ReplayGainReanalyze")).toInt();
+
+    if (analyzerEnabled) {
```

Comment:
This nested if is hard to read and even wrong:

if (analyzerEnabled && reanalyze) {
  // ignore stored replay gain
  return false;
} else {
  // use stored replay gain (if defined)
 return tio->getReplayGain().hasRatio();
}